powerdnsadmin/routes/domain.py

```diff
@@ -646,6 +691,43 @@ def change_account(domain_name):
         abort(500)
 
 
+def are_equal(old : RecordEntry, new: dict, domain_name: str):
+    if old.name == domain_name:
+        name_comparison = (new['record_name'] == '@')
+    else:
+        name_comparison = (old.name == new['record_name'] + "." + domain_name)
+
+    if not name_comparison or \
```

Comment:
In the openstack cli, each object inherits a to_dict() method that converts the object to a dictionnary. Wouldn't it be easier to implement a such function in RecordEntry and just replacing the name:

`dumped_record['name'] = '@' if dumped_record['name'] == domain_name else re.sub(r'\.' + re.escape(domain_name) + r"$")` ?

Then `if old != new: return False`